src/lib/rest/StringFilter.cpp

```diff
@@ -846,6 +892,7 @@ bool StringFilter::parse(const char* q, std::string* errorStringP)
     }
 
     str = NULL;  // So that strtok_r continues eating the initial string
+    item.patternValueToBeFreed = false;
```

Comment:
Why this explicit setting is needed? The StringFilterItem constructor at line L882 sets `patternValueToBeFreed` to `false`. What could happend between L882 and this one that "reset" that setting an needs to do it again?